apps/finance/app/src/components/LocalIdentityBadge/LocalIdentityBadge.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { Badge, IdentityBadge, font } from '@aragon/ui'
+import { LocalIdentityModalContext } from '../LocalIdentityModal/LocalIdentityModalManager'
+import { IdentityContext } from '../IdentityManager/IdentityManager'
+
+const LocalIdentityBadge = ({ address, ...props }) => {
+  const { resolve } = React.useContext(IdentityContext)
+  const { showLocalIdentityModal, updates$ } = React.useContext(
+    LocalIdentityModalContext
+  )
+  const [label, setLabel] = React.useState()
+  const handleResolve = async () => {
+    try {
+      const { name = null } = await resolve(address)
+      setLabel(name)
+    } catch (e) {
+      // address does not ressolve to identity
+    }
+  }
+  const handleClick = () => {
+    showLocalIdentityModal(address)
+      .then(handleResolve)
+      .catch(e => {
+        /* user cancelled modify intent */
+      })
+  }
+  React.useEffect(() => {
+    handleResolve()
+    const subscription = updates$.subscribe(updatedAddress => {
+      if (updatedAddress.toLowerCase() === address.toLowerCase()) {
+        handleResolve()
+      }
+    })
+    return () => subscription.unsubscribe()
+  }, [])
```

Comment:
Shouldn’t we re-run this if `address` changes?

Reply:
For a single IdentityBadge address won't change.

Reply:
I wonder if we shouldn’t still make sure our effects always define their dependencies even if the context doesn’t requires it? The same way we always clean up our events or timers, even for single life components, just to make it explicit and easier to move code. I know the React team plans to / has already implemented this fix in their [ESLint plugin for hooks](https://github.com/facebook/react/issues/14920) (by filling this array automatically), so maybe we could anticipate.